### PR TITLE
cli: Add new duration format, 'ms'

### PR DESCRIPTION
**Pull request by flexibeast**
Addresses #19.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -144,6 +146,15 @@ pub fn fmt_duration(style: DurationStyle, secs: i64) -> String {
...
         },
+        DurationStyle::MS => {
+            let m = secs / 60;
+            let s = secs % 60;
+            if m > 0 {
+                format!("{}:{:02}", m, s)
+            } else {
+                format!("0:{:02}", s)
+            }
+        },
```
Comment (vincentdephily) [inline src/main.rs:157]: Both `if` branches output the same thing, just use the body of the first.

I find it a bit strange that you're not formatting hours. Merge times over 60 minutes are not uncommon, and displaying them as `123:45` doesn't seem very usable ? My idea of the new format was that it'd be the same as `Duration::HMS` but without ignoring leading zeros, so `format!("{}:{:02}:{:02}")` in all cases.

File: src/main.rs [hunk 2]
```diff
@@ -115,15 +115,17 @@ impl FromStr for Show {
...
 }
 impl FromStr for DurationStyle {
     type Err = String;
     fn from_str(s: &str) -> Result<Self, Self::Err> {
         match s {
             "hms" => Ok(DurationStyle::HMS),
+            "ms" => Ok(DurationStyle::MS),
             "s" => Ok(DurationStyle::S),
-            _ => Err("Valid values are 'hms', 's'.".into()),
+            _ => Err("Valid values are 'hms', 'ms', 's'.".into()),
```
Comment (vincentdephily) [inline src/main.rs:128]: Bit of bikeshedding: "ms" reads like "milliseconds" to me. What about `hms_fixed` or similar ?

**Comment by vincentdephily** [comment]
Looks good, bonus points for taking care of the docs too :) Tell me what you think about format and name comments.

I've pushed a fix to a recent unittest regression, you might want to rebase you branch on current master to make the tests pass today.

**Comment by vincentdephily** [comment]
If there's no more comment, I'll merge this commit and apply my suggestions.

**Comment by flexibeast** [comment]
Er. i was hoping to get to this today; i haven't got to it sooner because there's a lot of other stuff going on in my life, and i haven't had the headspace for this. But if you're happy to make the changes you suggested yourself, i'm certainly happy for you to do so.

**Comment by vincentdephily** [comment]
No rush, sorry, mainly wanted to check if the review comments suited. I have a couple other features in mind for a release in a week or two, so you have a bit more time to polish this PR yourself, only if you want to :)

**Comment by flexibeast** [comment]
Thanks, yes, my preference would be to polish this PR. :-) i'll try to get to doing so in the next 24 hours.

**Comment by flexibeast** [comment]
i've rebased and locally made the changes you suggested. However, `cargo test` now produces a number of failures:
```
failures:
    commands::tests::exit_status
    commands::tests::log
    commands::tests::negative_merge_time
    commands::tests::stats
    commands::tests::stats_grouped
    commands::tests::timezone
```

And building and running `emlop` results in:
```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error { message: "\u{1b}[1;31merror:\u{1b}[0m The argument 'date' wasn't found", kind: ArgumentNotFound, info: Some(["date"]) }', src/main.rs:180:58
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
zsh: IOT instruction (core dumped)  emlop stats llvm
```

**Comment by vincentdephily** [comment]
Hum, master tests are [ok in the CI](https://github.com/vincentdephily/emlop/actions/runs/1777599308), something must have broken on your side but not sure what. The error message mentions a missing `date` argument, which is a new argument that was added after the `duration` argument in the code. My guess is that the `date` changes interfered with the `duration` changes during rebase.

Check with `git log -p` and/or `git diff` that your changes to the `duration` argument in `cli.rs` look correct. If you can't figure it out, just add your WIP to this PR and I'll take a look.

**Comment by flexibeast** [comment]
Gah, i'd stuffed up resolving a merge conflict during rebasing - sorry for the noise! Fixed that, tests passing, PR updated.

**Comment by vincentdephily** [comment]
Looks perfect. Thank you :slightly_smiling_face:

**Comment by flexibeast** [comment]
You're welcome - thanks for helping me contribute. :-)

**Pull request merged**